docs/book/v1/authorization-server.md

```diff
@@ -20,6 +20,27 @@ use Mezzio\Authentication\OAuth2;
 
 $app->post('/oauth2/token', OAuth2\TokenEndpointHandler::class);
 ```
+## Parsing JSON payloads in the token endpoint
+To enable the token endpoint to handle the POST requests in JSON, the [Body Parsing Middleware](https://docs.mezzio.dev/mezzio/v3/features/helpers/body-parse/) helper must be included in the application. 
+
+For example:
+
+```php
+use Mezzio\Helper\BodyParams\BodyParamsMiddleware;
+
+$app->pipe(BodyParamsMiddleware::class);
+```
+Generally speaking, we recommend AGAINST piping the body parsing middleware as generic middleware; instead, we recommend piping it into a route-specific pipeline:
+```php
+use Mezzio\Authentication\OAuth2;
+use Mezzio\Helper\BodyParams\BodyParamsMiddleware;
+
+$app->post('/oauth2/token', [
+    BodyParamsMiddleware::class,
+    OAuth2\TokenEndpointHandler::class
+], 'auth.token');
+```
+This ensures we only parse it when we are actually expecting a content body.
```

Comment:
The correct way of usage must be shown, not the wrong one.

``````suggestion
For example:

```php
use Mezzio\Authentication\OAuth2;
use Mezzio\Helper\BodyParams\BodyParamsMiddleware;

$app->post('/oauth2/token', [
    BodyParamsMiddleware::class,
    OAuth2\TokenEndpointHandler::class
], 'auth.token');
```

WARNING: Do not piping the body parsing middleware as generic middleware.
This ensures that the content body is only parsed when it is actually expected.
``````